### PR TITLE
[13.0] [mig] pos_loyalty

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Check pre-commit errors

**Comment by dsolanki-initos** [comment]
Pre-commit fails due to errors in tests.

**Comment by dsolanki-initos** [comment]
It would be great if we get the reviews here

File: pos_loyalty/views/res_partner_view.xml [hunk 1]
```diff
@@ -11,10 +11,11 @@
                     type="action"
                     name="%(point_of_sale.action_pos_pos_form)d"
                     context="{'search_default_partner_id': active_id,'default_partner_id': active_id}"
-                    attrs="{'invisible': ['|', ('customer', '=', False), ('pos_order_count', '=', 0)]}"
+                    attrs="{'invisible': ['|', ('customer_rank', '=', False), ('pos_order_count', '=', 0)]}"
```
Comment (rousseldenis) [inline pos_loyalty/views/res_partner_view.xml]: @dsolanki-initos  customer_rank is a Float

**Comment by ivantodorovich** [comment]
It seems pre-commit is still failing here

**Comment by dsolanki-initos** [comment]
Yes but it is due to test case

File: pos_loyalty/static/src/js/tests.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
+// Copyright 2004-2018 Odoo SA
+// Copyright 2018 Lambda IS DOOEL <https://www.lambda-is.com>
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
```
Comment (ivantodorovich) [inline pos_loyalty/static/src/js/tests.js]:
```suggestion
/* eslint-disable no-empty-function init-declarations */

```

To make pre-commit happy:

```
/home/travis/build/OCA/pos/pos_loyalty/static/src/js/tests.js
   33:33  error  Unexpected empty method 'run'                                 no-empty-function
   60:33  error  Unexpected empty method 'run'                                 no-empty-function
   70:33  error  Unexpected empty method 'run'                                 no-empty-function
   76:13  error  Variable 'i' should be initialized on declaration             init-declarations
   78:13  error  Variable 'current_char' should be initialized on declaration  init-declarations
  103:33  error  Unexpected empty method 'run'                                 no-empty-function
  130:33  error  Unexpected empty method 'run'                                 no-empty-function
  136:33  error  Unexpected empty method 'run'                                 no-empty-function
  149:29  error  Unexpected empty method 'run'                                 no-empty-function
  166:29  error  Unexpected empty method 'run'                                 no-empty-function
  171:29  error  Unexpected empty method 'run'                                 no-empty-function
  190:29  error  Unexpected empty method 'run'                                 no-empty-function
  202:29  error  Unexpected empty method 'run'                                 no-empty-function
```

**Comment by ivantodorovich** [comment]
Test are failing and this time is not because of pre-commit, can you check it out?
I also left some comments on the 14.0 PR, check if the same have to be applied here too

**Comment by hkapatel-initos** [comment]
> Test are failing and this time is not because of pre-commit, can you check it out?
> I also left some comments on the 14.0 PR, check if the same have to be applied here too

I have added changes as per suggestion but Travis is not happy because of https://app.travis-ci.com/github/OCA/pos/jobs/537261353#L1196.

**Comment by ivantodorovich** [comment]
> > Test are failing and this time is not because of pre-commit, can you check it out?
> > I also left some comments on the 14.0 PR, check if the same have to be applied here too
> 
> I have added changes as per suggestion but Travis is not happy because of https://app.travis-ci.com/github/OCA/pos/jobs/537261353#L1196.

That means the module's test tours are failing.

It's much easier to debug this locally than on travis, so I recommend you try to run the module tests locally and debug it :)

**Comment by dsolanki-initos** [comment]
@ivantodorovich @chienandalu Could you please review?

**Comment by dsolanki-initos** [comment]
@rafaelbn Can you please review this PR?

**Comment by dsolanki-initos** [comment]
Maintainers team, Could you please reopen this PR?

**Review by ivantodorovich — APPROVED** [review]

**Pull request closed without merging**